### PR TITLE
refactor(editor): Replace monaco-editor/prismjs with CodeMirror

### PR DESCRIPTION
**Pull request by netroy**
This is based on #4295 and #5446, but extracted into a separate PR to make reviewing #4295 easier, and to unblock #5517 and #5926

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
Bugs found so far:
1. ~~in `Execute Workflow` node, while editing the workflow json, the language is set to `js` instead of `json`~~
2. the code node is not getting the correct value passed in, so the node parameter `jsCode` gets resets to the placeholder value.

File: packages/editor-ui/src/components/CodeNodeEditor/theme.ts [hunk 1]
```diff
@@ -46,7 +50,10 @@ export const CODE_NODE_EDITOR_THEME = [
 			borderLeftColor: 'var(--color-code-caret)',
 		},
 		'&.cm-focused .cm-selectionBackgroundm .cm-selectionBackground, .cm-content ::selection': {
-			backgroundColor: 'var(--color-code-selection)',
+			// backgroundColor: 'var(--color-code-selection)',
```
Comment (netroy) [inline packages/editor-ui/src/components/CodeNodeEditor/theme.ts:53]: for some reason background-color on ::selection makes selection transparent on chrome (on linux). that's why I had to disable this. but, I'll revert this now, as the bug doesn't seem to exist on other platforms.

**Review by alexgrozav — DISMISSED** [review]
Awesome work!

**Review by alexgrozav — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.226.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.226.1)

**Pull request merged**